packages/strapi-design-system/src/NumberInput/NumberInput.js

```diff
@@ -150,7 +144,7 @@ export const NumberInput = React.forwardRef(
 
     const handleFocus = () => {
       if (value !== undefined) {
-        setInputValue(String(numberParserRef.current.parse(inputValue)));
+        setInputValue(String(numberParserRef.current.parse(inputValue) ?? INITIAL_VALUE));
```

Comment:
This fixes https://github.com/strapi/design-system/issues/654, since `0` is a falsy value.